src/hooks/useSettings.ts

```diff
@@ -0,0 +1,50 @@
+import { atom, useAtom } from "jotai";
+import { useEffect } from "react";
+
+export type Settings = {
+  title: string;
+  key: string;
+  type: string;
+  group: string;
+  value: any;
+  setValue: any;
+};
+
+const nodeDelayDefault = atom(0);
+const maximumRuntimeDefault = atom(3000);
+const settingsListDefault = atom([{} as Settings]);
+
+export const useSettings = () => {
+  const [nodeDelay, setNodeDelay] = useAtom(nodeDelayDefault);
+  const [maximumRuntime, setMaximumRuntime] = useAtom(maximumRuntimeDefault);
+  const [settingsList, setSettingsList] = useAtom(settingsListDefault);
+  useEffect(() => {
+    console.log("here");
+    setSettingsList([
+      {
+        title: "Node Delay (seconds)",
```

Comment:
We can move these constant values to `settingListDefault` variable and only update the necessary fields here.